### ISSUE TITLE
tx1-2: notice type handling

### ISSUE THREAD

**Issue by costezki** [OP-TED/ted-rdf-mapping#238]
[N Muric]

> In notices covering Contract Award Notices and other types of notices, the mappings are made in each case to the Result Notice which creates a wrong mapping to the other types of notices

**Comment by costezki** [comment]
In F21, F22, F23 Forms, we will use the `type` attribute in <notice/> element when present.

The following correspondence rules will be used
award_contarct -> epo:ResultNotice
contract -> epo:CompeitionNotice
concession_award_contract -> epo:ResultNotice
pri_call_competition -> epo:CompeitionNotice
pri -> epo:CompeitionNotice


![image](https://user-images.githubusercontent.com/13567133/203508333-2e4a6cd7-71b9-45f1-8de4-25e36c08e6e0.png)

![image](https://user-images.githubusercontent.com/13567133/203507658-0ad8ebde-e9c2-4579-8652-ae329b4d9909.png)

**Comment by csnyulas** [comment]
AFAICT, Qualification systems can be recognized from the presence of one of the following XPaths:
`LEFTI/QUALIFICATION`,
`OBJECT_CONTRACT/OBJECT_DESCR/QS/`, 
`OBJECT_CONTRACT/OBJECT_DESCR/RENEWAL`, and
`OBJECT_CONTRACT/OBJECT_DESCR/RENEWAL_DESC`

Is checking for this XPaths is sufficient to identify if a notice is regarding a Qualification System? 
Can it happen that all these fields are missing and still be a notice about a QS? 
Is there a better, more secure, way to identify that a notice is a QS, based on the information available in the XML, than checking for these XPaths? The `TYPE` attribute of the `NOTICE` element does not seem to give us any indication about this.

Is there a specific class, or a single (dedicated) property in EPO that can be used to identify a QS, **or** it is checked through the presence of any of these properties: `epo:hasQualificationSystemDuration`, `epo:hasQualificationSystemPeriod`, `epo:hasQualificationSystemRenewalDescription` ?

**Comment by csnyulas** [comment]
Also the epo:Notice created from form F20 will get a second type `epo:ContractModificationNotice`

**Comment by csnyulas** [comment]
Other Notice type values to be mapped (all collected from our sample data):
```
Form F01:
<NOTICE TYPE="PRI_CALL_COMPETITION"/>
<NOTICE TYPE="PRI_ONLY"/>
<NOTICE TYPE="PRI_REDUCING_TIME_LIMITS"/>

Form F04:
<NOTICE TYPE="PER_CALL_COMPETITION"/>
<NOTICE TYPE="PER_ONLY"/>
<NOTICE TYPE="PER_REDUCING_TIME_LIMITS"/>

Form F07:
<NOTICE TYPE="QSU_CALL_COMPETITION"/>

Form F21:
<NOTICE TYPE="AWARD_CONTRACT"/>
<NOTICE TYPE="CONTRACT"/>
<NOTICE TYPE="PRI_CALL_COMPETITION"/>

Form F22:
<NOTICE TYPE="AWARD_CONTRACT"/>
<NOTICE TYPE="CONTRACT"/>

Form F23:
<NOTICE TYPE="CONCESSION_AWARD_CONTRACT"/> 
<NOTICE TYPE="PRI"/>
```

**Comment by csnyulas** [comment]
Closing this, as it has been folly fixed. Opened an issue, #268, about implementing a potential improvement (related to performance an manageability) on the current solution.

**Comment by muricna** [comment]
In competition Notices according to the ontology the CompetitionNotice announcesRole and the ResultNotice refersToRole or in other cases announcesRole this is not reflected in the mapping.  The same can be said for announcesProcedure and refersToProcedure. 
The ontology should be informed that the Notice refersToProcedure predicate should be from ResultNotice
This comment applies to all of section I and it is up to the contractor to ensure the other sections are correct for other forms and for other classes such as Lot

**Comment by csnyulas** [comment]
Related to the last comment from @muricna: I created a separate issue for it (#326), because the problem described in the comment is different from what this issue was about. This issue is about attaching the appropriate Notice subclasses as a secondary `rdf:type` statement to the Notice instance. That has been fixed and tested, as can be seen in the linked commits. This is why we closed this ticket.
We will mark again this issue as Fixed, and further improvements/fixes will be addressed in issue #326.
